src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
Don't use `p`  or `p1` - i don't understand what it is. Your code will be read by others developers so it is important to write correct names for everything.

For better readability
```
const isMother = (mother) => people.find((person) => p1.mother === mother.name))

const women = withChildren
    ? people.filter((person) => person.sex === 'f'
      && isMother(person))
    : people.filter((person) => person.sex === 'f');

```